modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportStartReindexJobAction.java

```diff
@@ -56,35 +52,26 @@
 
 import static org.elasticsearch.index.reindex.ReindexTask.REINDEX_ORIGIN;
 
-public class TransportStartReindexJobAction
-    extends TransportMasterNodeAction<StartReindexJobAction.Request, StartReindexJobAction.Response> {
+public class TransportStartReindexJobAction extends HandledTransportAction<StartReindexJobAction.Request, StartReindexJobAction.Response> {
 
+    private final ThreadPool threadPool;
+    private final ClusterService clusterService;
     private final PersistentTasksService persistentTasksService;
     private final Client taskClient;
 
     @Inject
     public TransportStartReindexJobAction(TransportService transportService, ThreadPool threadPool,
-                                          ActionFilters actionFilters, IndexNameExpressionResolver indexNameExpressionResolver,
-                                          ClusterService clusterService, PersistentTasksService persistentTasksService, Client client) {
-        super(StartReindexJobAction.NAME, transportService, clusterService, threadPool, actionFilters, StartReindexJobAction.Request::new,
-            indexNameExpressionResolver);
+                                          ActionFilters actionFilters, ClusterService clusterService,
+                                          PersistentTasksService persistentTasksService, Client client) {
+        super(StartReindexJobAction.NAME, transportService, actionFilters, StartReindexJobAction.Request::new);
+        this.threadPool = threadPool;
+        this.clusterService = clusterService;
         this.persistentTasksService = persistentTasksService;
         this.taskClient = new OriginSettingClient(client, REINDEX_ORIGIN);
     }
 
     @Override
-    protected String executor() {
-        return ThreadPool.Names.SAME;
-    }
-
-    @Override
-    protected StartReindexJobAction.Response read(StreamInput in) throws IOException {
-        return new StartReindexJobAction.Response(in);
-    }
-
-    @Override
-    protected void masterOperation(Task task, StartReindexJobAction.Request request, ClusterState state,
-                                   ActionListener<StartReindexJobAction.Response> listener) {
+    protected void doExecute(Task task, StartReindexJobAction.Request request, ActionListener<StartReindexJobAction.Response> listener) {
         // TODO: If the connection is lost to the master, this action might be retried creating two tasks.
```

Comment:
Is this todo still accurate? I think this works OK now, since the REST action will call this on local node and thus there can be no retries?